src/Service.php

```diff
@@ -516,6 +516,9 @@ public function processUpdateQueue($limit = 200) {
                 $object = Concrete::getById($objectId);
                 if($object) {
                     $this->doUpdateIndexData($object);
+                } else {
+                    // Object no longer exists, remove from queue
```

Comment:
hmm, why not delete item in this case? not just update it?

Reply:
Good question, this was actually one of the things I planned to ask you about, but I forgot between the time I prepared the patch and submitted the PR!  :) didn't find any other place in the code where you deleted items from the queue table, so I went for what seemed the most safe solution, but I would guess that deleting them should be OK here, and you seem to agree with that, so I've updated my PR.